src/main/java/ru/ayaz/UserMessageDistributor.java

```diff
@@ -0,0 +1,63 @@
+package ru.ayaz;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+
+import static ru.ayaz.MessageParser.isCommand;
+
+public class UserMessageDistributor implements Runnable {
+
+    private Map<String, User> userMap;
+    private Map<String, UserSocketHandler> userSocketHandlerMap;
+    private BlockingQueue<UserMessage> messageQueue;
+    private MessageBroadcaster broadcaster;
+    private CommandExecutor executor;
+
+    UserMessageDistributor(HashMap<String, User> userMap, HashMap<String, UserSocketHandler> userSocketHandlerMap,
```

Comment:
Well, yeah, DI is good and all, but a map of objects is not exactly an object in itself. It looks kind of frustrating to see a map being passed around like that in different objects.
It tells us something about this map of users actually. It tells us that most likely this map of users is some kind of object or a part of some object.
As I described above, I believe it to be a part of the Room object.

DI is a good pattern, but there are some cases when you don't need to inject dependencies to test your object's functionality. In this particular case I consider a map to be just a data holder - a piece of memory which helps me implement some data storing functionality. When we are talking about DI, each dependency has the **behavior** your Object depends on. In case of Map this behavior is just read/write data in RAM. Its not a high level behavior one would want to mock. The Object itself depends on this behavior (each object has a state it needs to save/retrieve) - that's why data structures that the object is based on can be considered as a part of the object itself and not as its dependencies. - That's my point of view on it. If your data structure for storing object's state was a database service, then, of course you would want to mock it. But then again - database service is probably a dependency and not a part of object's state.
I guess there is really no visible distinction between when to inject a dependency and when not to.